kie-soup-maven-utils/kie-soup-maven-integration/src/main/java/org/appformer/maven/integration/InJarArtifactResolver.java

```diff
@@ -0,0 +1,169 @@
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.appformer.maven.integration;
+
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.URL;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import java.util.zip.ZipEntry;
+import java.util.zip.ZipInputStream;
+
+import org.apache.maven.project.MavenProject;
+import org.appformer.maven.integration.embedder.EmbeddedPomParser;
+import org.appformer.maven.integration.embedder.MavenProjectLoader;
+import org.appformer.maven.support.AFReleaseId;
+import org.appformer.maven.support.AFReleaseIdImpl;
+import org.appformer.maven.support.DependencyFilter;
+import org.eclipse.aether.artifact.Artifact;
+import org.eclipse.aether.artifact.DefaultArtifact;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import static java.util.Collections.emptyList;
+import static java.util.stream.Collectors.toList;
+
+public class InJarArtifactResolver extends ArtifactResolver {
+
+    private static final Logger log = LoggerFactory.getLogger(InJarArtifactResolver.class);
+
+    private ClassLoader classLoader;
+    private List<URL> jarRepository;
+    private PomParser pomParser;
+
+    InJarArtifactResolver(ClassLoader classLoader, AFReleaseId releaseId) {
+        this.classLoader = classLoader;
+        this.jarRepository = new ArrayList<>();
+        init(releaseId);
+    }
+
+    public boolean isLoaded() {
+        return pomParser != null;
+    }
+    // initialize in jar repository
+    private void init(AFReleaseId releaseId) {
+        jarRepository = buildResources();
+        pomParser = buildPomParser(releaseId);
+    }
+
+    private List<URL> buildResources() {
+        URL resourceURL = this.classLoader.getResource("");
+        if (resourceURL == null) {
+            return emptyList();
+        }
+        List<URL> resources = new ArrayList<>();
+        try (InputStream is = resourceURL.openStream();
+                ZipInputStream stream = new ZipInputStream(is);) {
+
+            ZipEntry entry = null;
+            while ((entry = stream.getNextEntry()) != null) {
+                if (isInJarFolder(entry.getName())) {
+                    resources.add(classLoader.getResource(entry.getName()));
+                }
+            }
+            if (!resources.isEmpty()) {
+                log.info("Found in jar repository {}", jarRepository);
```

Comment:
It also could be logged when resources is empty as a warning

Reply:
I set as alwasy logged.